lib/permutations.rb

```diff
@@ -1,4 +1,21 @@
 
 def permutations?(string1, string2)
-  
+  l1 = string1.length
+  l2 = string2.length
+  return false if l1 != l2
+
+  hash = {}
+  i = 0
+  while i < l1
+    hash[string1[i]] = true
+    i += 1
+  end
```

Comment:
While the `while` loops definitely works, another way to do this that follows Ruby style a bit more could be:
```ruby
string1.chars.each do |char|
  hash[char] = true
end
```

Reply:
Even better would be `string1.each_char`, which another student reminded me of :-)